ecr-login/ecr.go

```diff
@@ -51,7 +52,23 @@ func (self ECRHelper) Get(serverURL string) (string, string, error) {
 		return "", "", credentials.NewErrCredentialsNotFound()
 	}
 
-	client := self.ClientFactory.NewClientFromRegion(registry.Region)
+	var client api.Client
+	if registry.Fips {
+		resolver := endpoints.DefaultResolver()
+		endpoint, err := resolver.EndpointFor("ecr-fips", registry.Region)
+		if err != nil {
```

Comment:
Let's ignore this `err`.  The `EndpointFor` function returns an error because "ecr-fips" is not a known service, but it still generates a valid endpoint.